src/bundle/Controller/ContentDraftController.php

```diff
@@ -92,13 +92,18 @@ public function removeAction(Request $request): Response
             $result = $this->submitHandler->handle($form, function (ContentRemoveData $data) {
                 foreach (array_keys($data->getVersions()) as $version) {
                     $versionId = VersionId::fromString($version);
-
-                    $this->contentService->deleteVersion(
-                        $this->contentService->loadVersionInfoById(
-                            $versionId->getContentId(),
-                            $versionId->getVersionNo()
-                        )
-                    );
+                    $contentInfo = $this->contentService->loadContentInfo($versionId->getContentId());
+
+                    if (1 === count($this->contentService->loadVersions($contentInfo))) {
```

Comment:
If deletion of the last version should result in deletion of the content maybe we should reflect this in PHP API?

Reply:
We have this behavior in `content-forms` as well. Regardless, wouldn't it be a BC break?

Reply:
> If deletion of the last version should result in deletion of the content maybe we should reflect this in PHP API? Because now we will have such behavior only here

There's the following logic for `ContentService::deleteVersion`:
```php
        if (count($versionList) === 1 && !$versionInfo->isDraft()) {
            throw new BadStateException(
                '$versionInfo',
                'The Version is the last version of the Content item and cannot be removed'
            );
        }
```
If indeed what you're deleting here is a draft and it leaves a Content without any versions, then we probably miss some logic in `ContentService::deleteVersion`, since the intent was to allow deleting a single draft.

Reply:
Okay, in this case follow-up PRs will be made in `ezplatform-kernel` and `content-forms`.